src/main/java/tc/oc/pgm/commands/ModerationCommands.java

```diff
@@ -27,7 +39,36 @@ public static void report(
       MatchPlayer matchPlayer,
       Match match,
       Player player,
-      @Text String reason) {
+      @Text String reason)
+      throws CommandException {
+    if (!commandSender.hasPermission(Permissions.STAFF) && commandSender instanceof Player) {
+      // Check for cooldown
+      Instant lastReport = LAST_REPORT_SENT.getIfPresent(matchPlayer.getId());
+      if (lastReport != null) {
+        Duration timeSinceReport = Duration.between(lastReport, Instant.now());
+        long secondsRemaining = REPORT_COOLDOWN_SECONDS - timeSinceReport.getSeconds();
+        if (secondsRemaining > 0) {
+          Component secondsComponent = new PersonalizedText("" + secondsRemaining);
```

Comment:
I would use `Integer.toString(secondsRemaining)`, otherwise looks good to me.

Reply:
Oh snap, I did not see your suggestion! I’m adjusting this right now. 👍